BraveRewardsTests/LedgerDatabaseTests.mm

```diff
@@ -144,6 +147,638 @@ - (void)testNonZeroExcludedCount
+}
+
+#pragma mark - Media Publisher Info
+
+- (void)testInsertOrUpdateMediaPublisherInfoWithMediaKey
+{
+  [self createBATPublisherInfo:@"1" reconcileStamp:40 percent:0 createActivityInfo:true];
+  
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"MediaPublisherInfo"]);
+  
+  [self waitForFailedSaveAttempt:^{
+  [BATLedgerDatabase insertOrUpdateMediaPublisherInfoWithMediaKey:@"" publisherID:@"1"];
+  }];
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"MediaPublisherInfo"]);
+  
+  // Empty publishser
+  [self waitForFailedSaveAttempt:^{
+      [BATLedgerDatabase insertOrUpdateMediaPublisherInfoWithMediaKey:@"key" publisherID:@""];
+  }];
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"MediaPublisherInfo"]);
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+      [BATLedgerDatabase insertOrUpdateMediaPublisherInfoWithMediaKey:@"key" publisherID:@"1"];
+  }];
+  
+  XCTAssert([self countMustBeEqualTo:1 forEntityName:@"MediaPublisherInfo"]);
+}
+
+- (void)testMediaPublisherInfoWithMediaKey
+{
+  const auto publisherId = @"1";
+  const auto key = @"key";
+  
+  [self createBATPublisherInfo:@"1" reconcileStamp:40 percent:0 createActivityInfo:true];
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase insertOrUpdateMediaPublisherInfoWithMediaKey:key publisherID:publisherId];
+  }];
+  
+  const auto result = [BATLedgerDatabase mediaPublisherInfoWithMediaKey:key];
+  XCTAssertEqual(result.id, publisherId);
+  
+  const auto emptyResult = [BATLedgerDatabase mediaPublisherInfoWithMediaKey:@"empty"];
+  XCTAssertNil(emptyResult);
+}
+
+#pragma mark - Recurring Tips
+
+- (void)testInsertOrUpdateRecurringTipWithPublisherID
+{
+  const auto publisherId = @"1";
+  const auto amount = 20.0;
+  const auto now = [[NSDate date] timeIntervalSince1970];
+  
+  [self createBATPublisherInfo:@"1" reconcileStamp:40 percent:0 createActivityInfo:true];
+  
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"RecurringDonation"]);
+  
+  // Empty publisher id
+  [self waitForFailedSaveAttempt:^{
+    [BATLedgerDatabase insertOrUpdateRecurringTipWithPublisherID:@"" amount:amount dateAdded:now];
+  }];
+   
+  [self createRecurringTip:publisherId amount:amount date:now];
+  
+  XCTAssert([self countMustBeEqualTo:1 forEntityName:@"RecurringDonation"]);
+}
+
+- (void)testRecurringTipsForMonth
+{
+  const auto publisherId = @"1";
+  
+  const auto dateFormatter = [[NSDateFormatter alloc] init];
+  [dateFormatter setDateFormat:@"yyyy-MM-dd"];
+  
+  const auto date = [[dateFormatter dateFromString:@"2019-10-07"] timeIntervalSince1970];
+  const auto date2 = [[dateFormatter dateFromString:@"2020-11-17"] timeIntervalSince1970];
+  
+  [self createBATPublisherInfo:@"1" reconcileStamp:40 percent:0 createActivityInfo:true];
+  
+  [self createRecurringTip:publisherId amount:15.0 date:date];
+  [self createRecurringTip:@"2" amount:100.0 date:date2];
+  
+  XCTAssert([self countMustBeEqualTo:2 forEntityName:@"RecurringDonation"]);
```

Comment:
Is there a reason to do both the DataController count assertion and the object query count assertion below?